documentation/content/consistency.md

```diff
@@ -10,6 +10,32 @@ are, and what you, as a user, can expect.
 
 <!-- TODO more doc links, explain replica vs bucket?, ... -->
 
+### Vespa and CAP
+
+Vespa may be considered a limited subset of AP under the [CAP theorem](https://en.wikipedia.org/wiki/CAP_theorem).
+
+Under CAP, there is a fundamental limitation of whether any distributed system can offer
+guarantees on consistency (C) or availability (A) in scenarios where nodes are partitioned (P)
+from each other. Since there is no escaping that partitions can and will happen, we talk
```

Comment:
The network is reliable until it isn't 😉 Note that "partition" is a somewhat loose concept, and it doesn't have to mean a full (or physical) severing of connections. For instance, it's not too long since we observed asymmetric partitions between hosts where host A could connect to host B but not vice versa. We've also recently seen partitions caused by intermittent network interface shutdowns. In essence, not requiring some form of partition tolerance would require a 100% available networking stack (not just the physical layers), which would require magical unicorn hardware and software 🦄✨

Some cozy bedside reading on the matter: https://aphyr.com/posts/288-the-network-is-reliable

Reply:
Yes, network problems occur frequently, but why is it that they don't cause consistency problems in practice? Because they are not "partitions" in the relevant sense: That two parties (subsets of nodes) of a size and composition sufficient to constitute a functioning system instance have internal connectivity and reachability from clients but *not* inter-connectivity and therefore continue to operate under the assumption that they are the party which is currently live. *That* is a very real problem in global systems but not within data centers.